Fix "cannot be named without a reference" build error

### Description

Building the `sit-onyx` run successful, but shows an error. 
We must ensure to have a clear build log without errors and warnings.
Otherwise this could create issues downstream and in the future.

### Reproduction steps

1. Run `pnpm build-only` in `packages/sit-onyx`

### Error message

> > sit-onyx@0.1.0-alpha.0 build-only /Users/carlej/Desktop/schwarzit-github/onyx/packages/sit-onyx
> > vite build
> 
> vite v5.0.12 building for production...
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> type-fest@2.19.0/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> ../storybook-utils/src/actions.ts:84:14 - error TS2742: The inferred type of 'withVModelDecorator' cannot be named without a reference to '.pnpm/> @storybook+types@7.6.10/node_modules/@storybook/types'. This is likely not portable. A type annotation is necessary.
> 
> 84 export const withVModelDecorator = <T>(
>                 ~~~~~~~~~~~~~~~~~~~
> 
> ✓ 11 modules transformed.
> 
> [vite:dts] Start generate declaration files...
> computing gzip size (1)...[vite:dts] Declaration files built in 1241ms.
> 
> dist/style.css  346.99 kB │ gzip: 244.53 kB
> dist/index.js     5.54 kB │ gzip:   2.20 kB
> dist/style.css  346.99 kB │ gzip: 244.53 kB
> dist/index.cjs    4.38 kB │ gzip:   1.92 kB
> ✓ built in 1.54s

--- a/packages/storybook-utils/src/actions.ts
+++ b/packages/storybook-utils/src/actions.ts
@@ -1,12 +1,8 @@
 import { useArgs } from "@storybook/preview-api";
-import type { ArgTypes, Meta } from "@storybook/vue3";
+import type { ArgTypes, Decorator, Meta } from "@storybook/vue3";
 import { deepmerge } from "deepmerge-ts";
 import { isReactive, reactive, watch } from "vue";
-import type {
-  ArrayElement,
-  DefineStorybookActionsAndVModelsOptions,
-  ExtractVueEventNames,
-} from ".";
+import type { DefineStorybookActionsAndVModelsOptions, ExtractVueEventNames } from ".";
 
 /**
  * Utility to define Storybook meta for a given Vue component which will take care of defining argTypes for
@@ -31,14 +27,14 @@ import type {
  */
 export const defineStorybookActionsAndVModels = <T>(
   options: DefineStorybookActionsAndVModelsOptions<T>,
-): Meta<T> => {
+): Meta => {
   const defaultMeta = {
     argTypes: {
       ...defineActions(options.events),
       ...{}, // this is needed to fix a type issue
     },
     decorators: [withVModelDecorator(options.events)],
-  } satisfies Meta<T>;
+  } satisfies Meta;
 
   return deepmerge(options, defaultMeta);
 };
@@ -81,9 +77,7 @@ export const defineActions = <T>(events: ExtractVueEventNames<T>[]): ArgTypes =>
  * }
  * ```
  */
-export const withVModelDecorator = <T>(
-  events: ExtractVueEventNames<T>[],
-): ArrayElement<Meta<T>["decorators"]> => {
+export const withVModelDecorator = <T>(events: ExtractVueEventNames<T>[]): Decorator => {
   return (story, ctx) => {
     const vModelEvents = events.filter((event) => event.startsWith("update:"));
     if (!vModelEvents.length) return story();

--- a/packages/storybook-utils/src/types.ts
+++ b/packages/storybook-utils/src/types.ts
@@ -46,8 +46,3 @@ export type DefineStorybookActionsAndVModelsOptions<T> = Meta<T> & {
   component: NonNullable<T>;
   events: ExtractVueEventNames<T>[];
 };
-
-/**
- * A utility type that gets the type of an array element.
- */
-export type ArrayElement<A> = A extends readonly (infer T)[] ? T : never;

